build/Extensions/DepsJsonExtensions.cs

```diff
@@ -68,28 +73,136 @@ public static void RemoveOpenTelemetryLibraries(this JsonObject depsJson)
         }
     }
 
+    public static void RemoveOpenTelemetryLibraries(this JsonObject depsJson)
+    {
+        RemoveLibrary(depsJson, lib => lib.StartsWith("OpenTelemetry", StringComparison.Ordinal));
+    }
+
+    public static async Task CleanDuplicatesAsync(this JsonObject depsJson)
+    {
+        var map = DependencyAnalyzer.BuildDependencyMap(depsJson);
+
+        var result = AnalyzeAdapterDependencies(
+            // TODO: Scan these packages from OpenTelemetry.AutoInstrumentation.AdditionalDeps
+            await NugetPackageHelper.GetPackageDependenciesAsync(MongoClientIntegrationMetadata.AdapterAssembly, "1.3.0"),
+            await NugetPackageHelper.GetPackageDependenciesAsync(
+                MongoClientIntegrationMetadata.InstrumentedAssembly,
+                MongoClientIntegrationMetadata.MinimumVersion,
+                MongoClientIntegrationMetadata.MaximumVersion)
+        );
+
+        foreach (var framework in result)
+        {
+            var latestVersion = framework.Value
+                .OrderByDescending(x => x.Key.Version)
+                .FirstOrDefault();
+
+            if (latestVersion.Value == null)
+            {
+                continue;
+            }
+
+            foreach (var dependency in latestVersion.Value)
+            {
+                // All package versions contain the same optimisable dependency
+                var canBeOptimized = framework.Value
+                    .All(x => x.Value.Contains(dependency));
+                if (!canBeOptimized)
+                {
+                    continue;
+                }
+
+                // Remove the main library
+                RemoveLibrary(depsJson, lib => lib.Equals(dependency));
+
+                // Remove transient leftovers
+                DependencyAnalyzer.Cleanup(map, dependency)
+                    .ForEach(transient =>
+                        RemoveLibrary(depsJson, lib => lib.Equals(transient)));
+
+                // TODO: This is just cleaning up deps json, we need to manually
+                // clean shared store also?
+            }
+        }
+    }
+
+    private static Dictionary<NuGetFramework, Dictionary<NuGetVersion, ICollection<string>>> AnalyzeAdapterDependencies(
```

Comment:
@pjanotti I think this can be simplified a lot by removing .net framework info 🤔 
for .net6 we know that packages with .net6 or latest netstandard apply
for .net7 we know that packages with .net7 or latest netstandard apply

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2294/commits/b4f97c24fdf5d2c7ed3f54b1057e90b740bb0cfe